packages/excalidraw/components/App.tsx

```diff
@@ -3006,6 +3006,7 @@ class App extends React.Component<AppProps, AppState> {
         const imageElement = this.createImageElement({ sceneX, sceneY });
         this.insertImageElement(imageElement, file);
         this.initializeImageDimensions(imageElement);
+        this.store.shouldCaptureIncrement();
```

Comment:
Thanks for the submission! This is good first step, though probably not enough, reasons:

1. there are several `insertImageElement` usages, likely covering different use cases in addition to simple paste (didn't check in detail); each of those should be manually tested and possibly captured with history - those that should be captured should have a respective tests in `history.test.tsx`, those that shouldn't be captured, should have a comment why they shouldn't be captured
2. images are kind of a special type of element, as they are loaded async, so we really have to make sure that we capture in the history just what we want (i.e. any "concurrent" update performed in the meantime, such as from collab, or from a different action, should not be included in the captured increment)
3. I am sure this regressed (at least partially) as part of https://github.com/excalidraw/excalidraw/pull/7348, as during the PR some deployments behaved similarly as your PR, though I am not sure why, as it seems we never capture increments (~recorded changes) here. We should check how the history with images worked before the PR, how it worked in some deployments during PR, and how it regressed, so it won't regress in the future - I can hopefully help here a bit.

Also, atm there is a bug when pasting multiple times (could be related to point 2.):

https://github.com/user-attachments/assets/ef551361-b375-47f7-aa55-a4ed42b1a58e

Reply:
Thank you for your reply amidst your busy schedule. I will try to see if I can find a solution to this problem.